python/src/linalg.cpp

```diff
@@ -210,4 +210,28 @@ void init_linalg(py::module_& parent_module) {
             array([[-2.23607, -3.57771],
                    [0, 0.447214]], dtype=float32)
       )pbdoc");
+  m.def(
+      "svd",
+      &svd,
+      "a"_a,
+      py::kw_only(),
+      "stream"_a = none,
+      R"pbdoc(
+        svd(a: array, *, stream: Union[None, Stream, Device] = None) -> (array, array, array)
+
+        The Singular Value Decomposition (SVD) of the input matrix.
+
+        This function supports arrays with at least 2 dimensions. When the input
+        has more than two dimensions, the function operates in "stacked mode" like
+        numpy.
+
+        Args:
+            a (array): Input array.
+            stream (Stream, optional): Stream or device. Defaults to ``None``
+              in which case the default stream of the default device is used.
+
+        Returns:
+            tuple(array, array, array): The ``U``, ``S``, and ``Vt`` matrices, such that
```

Comment:
Is it actually a `tuple` or is it a `list`?

Reply:
If it's a list (which I think is what pybind11 will cast the vector to),  I think it would be good to make it a `tuple` to stay consistent with NumPy.

Reply:
Yes sorry, I overlooked this when changing the C++ signature from `tuple` to `vector`.